libs/designsystem/modal/src/modal-navigation.service.ts

```diff
@@ -340,4 +340,15 @@ export class ModalNavigationService {
     }
     return parentRoute;
   }
+
+  handleBackButton(modal: HTMLIonModalElement) {
+    const popStateEvent$ = fromEvent(window, 'popstate');
```

Comment:
Consider only handling 1 `popstate` event:
```suggestion
    const popStateEvent$ = fromEvent(window, 'popstate').pipe(first());
```